src/Gaffer/ValuePlug.cpp

```diff
@@ -624,8 +626,24 @@ class ValuePlug::ComputeProcess : public Process
 			const ThreadState &threadState = ThreadState::current();
 			const Context *currentContext = threadState.context();
 
-			const ComputeNode *computeNode = IECore::runTimeCast<const ComputeNode>( p->node() );
-			const ComputeProcessKey processKey( p, plug, computeNode, computeNode ? computeNode->computeCachePolicy( p ) : CachePolicy::Uncached, precomputedHash );
+			CachePolicy cachePolicy = CachePolicy::Uncached;
+			if( computeNode )
```

Comment:
It seems odd that we'd be caching `setFrom()` when it happens to occur on a ComputeNode, but not when it doesn't - the type of node has no impact on the cacheworthiness of the conversion. I know this is what we were doing before, but I think that's just through an oversight, and it would be worth tidying this up now, ahead of a future consideration of a change in cache policy for such conversions. So I propose to use `Legacy` whenever we have an input, regardless of node type.